src/lib.rs

```diff
@@ -645,7 +630,7 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
         model: &mut KeystoreModel<'a, L, impl KeystoreBackend<'a, L>, Meta>,
         receipt: &TransactionReceipt<L>,
     ) -> Result<TransactionStatus, KeystoreError<L>> {
-        match self.txn_state.transactions.status(&receipt.uid) {
+        match model.transactions.get(&receipt.uid)?.status() {
```

Comment:
Honestly, I would be in favor of getting rid of this entire mechanism, and making `type TransactionReceipt<L> = TransactionHash<L>`. This has never really worked properly for foreign transactions, and anyway it leaks privacy to call this for a transaction you expect to receive. For our own transactions, meanwhile, we now have the transaction status saved and updated, so this can all be replaced with `keystore.transactions().get(hash).status()`.

If you really want to track a foreign transaction by its hash, you can use a block explorer for that.

Reply:
Yeah I will get rid of this